File: Easy/70. Climbing Stairs/official solution (2).java
// 70. Climbing Stairs

// Runtime: 0 ms, faster than 100.00% of Java online submissions for Climbing Stairs.

// Memory Usage: 36.5 MB, less than 5.26% of Java online submissions for Climbing Stairs.


public class Solution {
    // Recursion with Memoization
    public int climbStairs(int n) {
        int memo[] = new int[n + 1];
        return climbStairs(0, n, memo);
    }

    public int climbStairs(int curr, int dest, int memo[]) {
        if (curr > dest) {
            return 0;
        } else if (curr == dest) {
            return 1;
        } else if (memo[curr] > 0) {
            return memo[curr];
        }

        // Store the result at each step in `memo` array and directly returning the result from the `memo` array whenever that function is called again.
        memo[curr] = climbStairs(curr + 1, dest, memo) + climbStairs(curr + 2, dest, memo);
        return memo[curr];
    }
}

File: Easy/796. Rotate String/official solution (2).py
# 796. Rotate String

# Runtime: 28 ms, faster than 82.80% of Python3 online submissions for Rotate String.

# Memory Usage: 13.8 MB, less than 64.00% of Python3 online submissions for Rotate String.


class Solution:
    # Simple Check
    def rotateString(self, A: str, B: str) -> bool:
        return len(A) == len(B) and B in A + A